examples/promtracker/src/receiver.mo

```diff
@@ -0,0 +1,55 @@
+import Stream "../../../src/StreamReceiver";
+import Buffer "mo:base/Buffer";
+import Debug "mo:base/Debug";
+import Array "mo:base/Array";
+import Text "mo:base/Text";
+import PT "mo:promtracker";
+import HTTP "http";
+
+actor class Receiver() = self {
+  let metrics = PT.PromTracker("", 65);
+  metrics.addSystemValues();
+
+  let chunkSizes = metrics.addGauge("chunk_sizes", "", #both, Array.tabulate<Nat>(5, func(i) = 2 ** i), false);
+  let messageSizes = metrics.addGauge("message_sizes", "", #both, Array.tabulate<Nat>(5, func(i) = 2 ** i), false);
+
+  type ControlMessage = Stream.ControlMessage;
+  type ChunkMessage = Stream.ChunkMessage<?Text>;
+
+  let received = Buffer.Buffer<?Text>(0);
+
+  let receiver = Stream.StreamReceiver<?Text>(
+    func(index : Nat, item : ?Text) {
+      assert received.size() == index;
+      received.add(item);
+    },
+    null,
+  );
+
+  let length = metrics.addPullValue("sent", "", func() = receiver.length());
+
+  public shared func receive(message : ChunkMessage) : async ControlMessage {
+    let size = switch (message.1) {
+      case (#chunk a) a.size();
+      case (_) 0;
+    };
+    chunkSizes.update(size);
```

Comment:
This code runs regardless of whether the chunk is ok or not. If it ends up being a gap then we don't want to run it.